pkg/plugins/discovery.go

```diff
@@ -33,6 +33,10 @@ func Scan() ([]string, error) {
 			return nil
 		}
 
+		if fi.IsDir() {
```

Comment:
We allow nesting the socket inside subdirectories. 
From https://docs.docker.com/engine/extend/plugin_api/ : "You can define each plugin into a separated subdirectory if you want to isolate definitions from each other. For example, you can create the flocker socket under /run/docker/plugins/flocker/flocker.sock and only mount /run/docker/plugins/flocker inside the flocker container."

So dont do this.